client/webpack.config.js

```diff
@@ -259,8 +266,8 @@ const webpackConfig = {
 				debug: path.resolve( __dirname, '../node_modules/debug' ),
 				store: 'store/dist/store.modern',
 				gridicons$: path.resolve( __dirname, 'components/gridicon' ),
-				'@wordpress/data': require.resolve( '@wordpress/data' ),
-				'@wordpress/i18n': require.resolve( '@wordpress/i18n' ),
+				'@wordpress/data': findPackage( '@wordpress/data' ),
```

Comment:
This was originally failing with https://github.com/webpack/webpack/issues/12071

However, this was a suboptimal: by using `require.resolve` we are using Node resolution and it resolves to `node_modules/@wordpress/data/build/index.js` (a CJS compilation). The function `findPackage` will only return the absolute path to the package and let Webpack do the final resolution, which will resolve to `node_modules/@wordpress/data/build-module/index.js`, an ESM compilation.